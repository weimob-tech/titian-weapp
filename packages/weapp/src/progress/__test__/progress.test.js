const { getComponent } = require('../../common/test/getComponent');

describe('test component', () => {
  it('should render success', () => {
    const comp = getComponent('progress', {
      color: '#ff0000',
      gradientColor: {
        from: '#ffff00',
        to: '#ff0000'
      }
    });
    comp.triggerLifeTime('attached');
    expect(comp).toMatchSnapshot();
  });

  it('should render value success', () => {
    const comp = getComponent('progress', {
      color: '#ff0000',
      value: 50
    });
    comp.triggerLifeTime('attached');
    expect(comp).toMatchSnapshot();
  });
});
