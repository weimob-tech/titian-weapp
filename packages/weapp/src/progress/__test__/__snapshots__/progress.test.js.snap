// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component should render success 1`] = `
<main>
  <wx-view
    class="ext-class titian-progress"
  >
    <wx-view
      class="titian-progress-determinate"
      style="height:4rpx"
    >
      <wx-view
        class="titian-progress-bar-linear titian-progress-bar"
        style="background-color:#ff0000;background-image:linear-gradient(to right, #ffff00, #ff0000);color:#ff0000"
      />
      <wx-view
        class="titian-progress-assist-bar"
        style="background-color:#ff0000"
      />
    </wx-view>
  </wx-view>
</main>
`;

exports[`test component should render value success 1`] = `
<main>
  <wx-view
    class="ext-class titian-progress"
  >
    <wx-view
      class="titian-progress-determinate"
      style="height:4rpx"
    >
      <wx-view
        class="titian-progress-bar"
        style="transform:translateX(-50%);background-color:#ff0000"
      />
    </wx-view>
  </wx-view>
</main>
`;
